sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { CredentialUnavailableError } from "../client/errors";
+import { credentialLogger, formatSuccess, formatError } from "../util/logging";
+import { trace } from "../util/tracing";
+import { ensureValidScope, getScopeResource } from "../util/scopeUtils";
+import { processUtils } from "../util/processUtils";
+
+const logger = credentialLogger("AzurePowerShellCredential");
+
+/**
+ * Formats a command based on the platform executing it.
+ * @internal
+ */
+export function formatCommand(commandName: string): string {
+  if (process.platform === "win32") {
```

Comment:
I'm not sure this is necessary - I can run `powershell` or `pwsh` without the extension in windows. Might be able to simplify things

Reply:
Are you trying from the windows command line? I just tried and without .exe I get ENOENT.

Reply:
Yea I tried command line, powershell, developer command line for VS, and powershell core... does it work with the extension for you (just to confirm)?

Reply:
yep, I'll share screen!